tokio/src/util/blocking_check.rs

```diff
@@ -0,0 +1,25 @@
+#[cfg(unix)]
+use std::os::fd::AsFd;
+
+#[cfg(unix)]
+#[allow(unused_variables)]
+pub(crate) fn check_socket_for_blocking<S: AsFd>(s: &S) -> crate::io::Result<()> {
```

Comment:
Why does this return `io::Result`? It doesn't seem like the fn needs a return argument.

Reply:
If the nonblocking check fails, propating the error seems the most straightforward response.